test/functional/wallet_basic.py

```diff
@@ -121,13 +121,40 @@ def run_test(self):
         # Exercise locking of unspent outputs
         unspent_0 = self.nodes[2].listunspent()[0]
         unspent_0 = {"txid": unspent_0["txid"], "vout": unspent_0["vout"]}
+        # Trying to unlock an output which isn't locked should error
         assert_raises_rpc_error(-8, "Invalid parameter, expected locked output", self.nodes[2].lockunspent, True, [unspent_0])
+
+        # Locking an already-locked output should error
         self.nodes[2].lockunspent(False, [unspent_0])
         assert_raises_rpc_error(-8, "Invalid parameter, output already locked", self.nodes[2].lockunspent, False, [unspent_0])
+
+        # Restarting the node should clear the lock
+        self.restart_node(2)
+        self.nodes[2].lockunspent(False, [unspent_0], True)
+
+        # Restarting the node now the lock is stored in memory should keep the lock
```

Comment:
In 94b6c8db5010bf1f38f1458866575cb599d17326 "Update lockunspent tests for lock persistence"

nit:

```suggestion
        # Restarting the node with the lock written to the wallet should keep the lock
```